crates/rattler_networking/src/s3_middleware.rs

```diff
@@ -80,41 +83,62 @@ impl S3 {
+                    aws_sdk_s3::config::Builder::from(&sdk_config)
+                        .endpoint_url(endpoint_url)
+                        .region(aws_sdk_s3::config::Region::new(region))
+                        .force_path_style(force_path_style)
+                        .credentials_provider(aws_sdk_s3::config::Credentials::new(
+                            access_key_id,
+                            secret_access_key,
+                            session_token,
+                            None,
+                            "pixi",
+                        ))
+                }
                 (_, Some(_)) => {
-                    panic!("Unsupported authentication method"); // todo proper error message
+                    return Err(anyhow::anyhow!("Unsupported authentication method"));
+                }
+                (_, None) => {
+                    eprintln!("No authentication found");
+                    let sdk_config = aws_config::defaults(BehaviorVersion::latest())
+                        .no_credentials() // Turn off request signing
+                        .load()
+                        .await;
+                    aws_sdk_s3::config::Builder::from(&sdk_config)
+                        .endpoint_url(endpoint_url)
+                        .region(aws_sdk_s3::config::Region::new(region))
+                        .force_path_style(force_path_style)
                 }
-                (_, None) => todo!("should use no credentials provider and not sign"),
-                // (_, None) => aws_sdk_s3::config::Builder::from(&sdk_config)
-                //     .endpoint_url(endpoint_url)
-                //     .region(aws_sdk_s3::config::Region::new(region))
-                //     .force_path_style(force_path_style),
             };
-            let aws_config = config_builder.build();
-            aws_sdk_s3::Client::from_conf(aws_config)
+            let s3_config = config_builder.build();
+            Ok(aws_sdk_s3::Client::from_conf(s3_config))
         } else {
-            // TODO: infer path style from endpoint URL or other means and set
-            // .force_path_style(true) on builder if necessary
-            let config = aws_sdk_s3::config::Builder::from(&sdk_config).build();
-            aws_sdk_s3::Client::from_conf(config)
+            let sdk_config = aws_config::defaults(BehaviorVersion::latest()).load().await;
+            let mut s3_config_builder = aws_sdk_s3::config::Builder::from(&sdk_config);
+            // Infer if we expect path-style addressing from the endpoint URL.
+            if let Some(endpoint_url) = sdk_config.endpoint_url() {
+                // If the endpoint URL is localhost, we most likely have to use path-style addressing.
+                if endpoint_url.starts_with("http://localhost") {
+                    s3_config_builder = s3_config_builder.force_path_style(true);
+                }
```

Comment:
this seems overly specific; there are more cases where this happens; i would just add a comment to https://github.com/awslabs/aws-sdk-rust/issues/1230 here

Reply:
ah, i'm assuming you added this because otherwise the minio integration tests are failing, right?

Reply:
Yes, that's why I need it

Reply:
I still added an xref here.